### ISSUE TITLE
composer.v1.Environment gets replaced instead of updated and fails to re-deploy

### ISSUE THREAD

**Issue by ColeSiegelTR** [pulumi/pulumi-google-native#217]
Hi,

I'm trying to create and update an instance of `composer.v1.Environment`.

Certain properties such as `pypiPackages` within `softwareConfig` cannot be set initially and can only be set on update. However, I get this error when I try to update my existing environment resource with dependencies: `error: error sending request: googleapi: Error 400: User-provided PyPi dependencies forbidden at environment creation. Please create an environment without dependencies then perform an update.`

Also, Pulumi seems to delete and re-create the environment unnecessarily when configuration changes are made, which results in new DAG buckets being created as well as any airflow variables being wiped.

What is the logic behind whether an update should result in delete/re-create or in-place update?

**Comment by ColeSiegelTR** [comment]
Another comment about Composer replace

If I update certain properties e.g. the `imageVersion` then Pulumi attempts to delete and re-create the composer environment. However, the deletion does not wait until the resource is actually gone and the re-creation then fails as the name already exists. Then the stack is in a broken state where composer exists in the stack but not in GCP.

**Comment by viveklak** [comment]
Could this be related to #185 @mikhailshilkov?

**Comment by mikhailshilkov** [comment]
Yes, the 

> Pulumi seems to delete and re-create the environment unnecessarily when configuration changes are made

part is likely #185. Our replacement detection is somewhat broken at the moment.

**Comment by ColeSiegelTR** [comment]
Hi @mikhailshilkov thanks for fixing this!

What release will this be available in? I tested with 0.9.0 and see the same behavior.

**Comment by mikhailshilkov** [comment]
@ColeSiegelTR It's now available in 0.10.0

**Comment by ColeSiegelTR** [comment]
Still doesn't work, fails with error below

```
 ~   └─ google-native:composer/v1:Environment                   cloud_composer                                                             **updating failed**     [diff: +config];
 
Diagnostics:
    error: update failed
 
  google-native:composer/v1:Environment (cloud_composer):
    error: error sending request: googleapi: Error 400: Found 1 problem:
        1) Request must not contain an empty update_mask.
```
